File: 3_2/3_2.cs
using System;
using System.Collections.Generic;

namespace EvenDigitsCounter
{
    class Program
    {
        static int CountEvenDigits(int number)
        {
            int count = 0;
            while (number > 0)
            {
                int digit = number % 10;
                if (digit % 2 == 0)
                {
                    count++;
                }
                number = number / 10;
            }
            return count;
        }

        static void Main(string[] args)
        {
            int number;
            List<int> numbers = new List<int>();
            Console.WriteLine("Enter the numbers (enter 0 to complete):");
            while (true)
            {
                number = Convert.ToInt32(Console.ReadLine());
                if (number == 0)
                {
                    break;
                }
                numbers.Add(number);
            }
            foreach (int num in numbers)
            {
                int numEvenDigits = CountEvenDigits(num);
                Console.WriteLine("Number of even digits in " + num + ": " + numEvenDigits);
            }
        }
    }
}

File: 3_2/3_2.swift
import Foundation

func countEvenDigits(_ number: Int) -> Int {
    var count = 0
    var num = number
    while num > 0 {
        let digit = num % 10
        if digit % 2 == 0 {
            count += 1
        }
        num = num / 10
    }
    return count
}

func main() {
    var numbers = [Int]()
    print("Enter the numbers (enter 0 to complete):")
    while true {
        if let input = readLine(), let number = Int(input) {
            if number == 0 {
                break
            }
            numbers.append(number)
        }
    }
    for num in numbers {
        let numEvenDigits = countEvenDigits(num)
        print("Number of even digits in \(num): \(numEvenDigits)")
    }
}

main()
